### PR TITLE
Clean/shellcheck scripts; produce concatenated logs

### PR DESCRIPTION
**Pull request by eriknordmark**

### CODE REVIEW

File: scripts/generate-device.sh [hunk 1]
```diff
@@ -10,10 +10,10 @@
...
 if [ $# != 1 ]; then
-    myname=`basename $0`
+    myname=$(basename "$0")
     echo "Usage: $myname <output basename>"
     exit 1
 fi
-lifetime=`expr 365 \* 20`
-dir=`dirname $0`
-$dir/generate-self-signed.sh $lifetime $1
+lifetime=$(365 \* 20)
```
Comment (deitch) [inline scripts/generate-device.sh]: Might have an easier time with 

```sh
lifetime=$(( 365 * 20 ))
```

File: Makefile [hunk 2]
```diff
@@ -33,11 +33,9 @@ APPS1 = logmanager ledmanager downloader verifier client zedrouter domainmgr ide
 
 SCRIPTS = \
```
Comment (deitch) [inline Makefile]: I could not figure out where `$(SCRIPTS)` is used elsewhere in this `Makefile`. Do other makefiles `include` this one?

**Comment by deitch** [comment]
I love a commit whose comment is: "oops" :-)

File: scripts/device-steps.sh [hunk 3]
```diff
@@ -20,10 +20,9 @@ AGENTS="$AGENTS0 $AGENTS1"
 
 PATH=$BINDIR:$PATH
 
-STARTTIME=`date`
-echo "Starting device-steps.sh at" $STARTTIME
-echo "go-provision version:" `cat $BINDIR/versioninfo`
-echo "go-provision version.1:" `cat $BINDIR/versioninfo.1`
+echo $(date -Ins -u) "Starting device-steps.sh"
```
Comment (deitch) [inline scripts/device-steps.sh]: Does this work? `date -Ins -u` is not valid in `busybox` (on which `alpine` is based) or BSD-based distros.

File: scripts/generate-device.sh (hunk 1 above)
Comment (eriknordmark) [reply scripts/generate-device.sh]: Thanks

File: Makefile (hunk 2 above)
Comment (eriknordmark) [reply Makefile]: Good catch

File: scripts/device-steps.sh (hunk 3 above)
Comment (eriknordmark) [reply scripts/device-steps.sh]: It works fine for me on the device inside the container.
But we have
lrwxrwxrwx 1 root root 20 Apr 17 02:33 /bin/date -> ../usr/bin/coreutils

**Comment by eriknordmark** [comment]
> I love a commit whose comment is: "oops" :-)

If I don't want to admit guilt I'll use 'commit -m . '  ;-)

File: scripts/device-steps.sh (hunk 3 above)
Comment (deitch) [reply scripts/device-steps.sh]: Ah ok. As long as it works there it is good.

**Comment by rvs** [comment]
This looks like a reasonable step forward (and I agree with @deitch comments)

I think the plan should be to commit this (perhaps squash it before committing?) and then start using yetus (cc @aw-zededa ) for a full shell vetting

Sounds good?

**Comment by aw-zededa** [comment]
> I think the plan should be to commit this (perhaps squash it before committing?) and then start using yetus (cc @aw-zededa ) for a full shell vetting

FWIW, I filed https://issues.apache.org/jira/browse/YETUS-853 to add the ability to fail PRs but not fail branches to go along with this plan. Given that 0.10.0 is a bit delayed for a few days anyway, it should show up in the release I plan on cutting over the next week (barring no other surprises).

**Pull request merged**